pinot-common/src/main/java/org/apache/pinot/common/utils/CommonConstants.java

```diff
@@ -320,6 +320,11 @@ public ServerType getServerType() {
         IN_PROGRESS, DONE
       }
 
+      public enum CompletionMode {
+        DEFAULT, // default behavior
+        DOWNLOAD // always download the segment
```

Comment:
please add more document the behavior here,as verbose as you can (comments are free :)). In the current implementation, we are planning on the server paying attention to this config, and deciding to download or build a realtime segment that it knows are equivalent either way.